source/TagRestrictions/TagsChecker.cs

```diff
@@ -122,25 +130,55 @@ private bool RequiredAnyCheck(HashSet<string> sourceTags, HashSet<string> target
             return false;
         }
 
-        private bool RequiredCheck(HashSet<string> sourceTags, HashSet<string> targetTags,
-            Func<string, IEnumerable<string>> requiredTags, ChassisLocations location = ChassisLocations.None)
+        private bool RequiredAnyCheck(HashSet<string> sourceTags, HashSet<string> targetTags,
+          Func<string, IEnumerable<string>> requiredTags, ChassisLocations location = ChassisLocations.None)
         {
             foreach (var tag in sourceTags)
-            foreach (var requiredTag in requiredTags(tag))
             {
-                if (targetTags.Contains(requiredTag)) continue;
+                var hasMetAnyRequiredTags = true;
```

Comment:
not sure if the comments previously here were useful or not, but something here would be useful